onnx/defs/schema.h

```diff
@@ -29,12 +29,17 @@ struct FunctionBodyBuildContext {
   virtual const AttributeProto* getAttribute(const std::string& name) const = 0;
   virtual bool hasInput(int i) const = 0;
   virtual bool hasOutput(int i) const = 0;
+  // getInputType(i) should return null for missing optional inputs, or if
+  // type-inference could not infer the input-type (erroneous model).
+  virtual const TypeProto* getInputType(int i) const = 0;
```

Comment:
rename i to say input_index for better readability

Reply:
I felt it would be simplest to maintain compatibility with the existing methods like hasInput/hasOutput. If we change it, we should do it for all simultaneously. (A related note: if we do change it, it would be more useful to change the type to unsigned as well; but then it starts becoming a more significant change ... something to think about.) Any thoughts on what exactly we should do?

Reply:
I did not realize other methods used the same convention. For this PR I suggest updating this and other methods to use a readable name instead of i ... and we can update the type to uint as part of another PR.

Reply:
Done